projects/dsp-ui/src/lib/viewer/services/upload-file.service.spec.ts

```diff
@@ -20,23 +21,54 @@ describe('UploadFileService', () => {
 
     beforeEach(() => {
         const initSpy = jasmine.createSpyObj('AppInitService', ['config']);
-        httpClientSpy = jasmine.createSpyObj('HttpClient', ['post']);
+        const httpClientSpyObj = jasmine.createSpyObj('HttpClient', ['post']);
```

Comment:
What about using [HttpTestingController](https://angular.io/api/common/http/testing/HttpTestingController)?

https://stackoverflow.com/questions/52312409/httptestingcontroller-expectone-verify-only-the-api-endpoint

Reply:
I think I got it to work in the past, it took my a while though.

Reply:
@tobiasschweizer good point. I'm trying to get this idea second go, but so far not going well.

Reply:
I think I found an example here: https://github.com/dasch-swiss/knora-ui/blob/master/projects/knora/core/src/lib/services/v2/ontology.service.spec.ts

This makes sure all HTTP requests have been handled: 

https://github.com/dasch-swiss/knora-ui/blob/2986f4e048daa28e5eb23b4920b4dea6cebebf77/projects/knora/core/src/lib/services/v2/ontology.service.spec.ts#L58-L61

For each HTTP request via the `HTTPClient`, you have to set up a thing like this:

https://github.com/dasch-swiss/knora-ui/blob/2986f4e048daa28e5eb23b4920b4dea6cebebf77/projects/knora/core/src/lib/services/v2/ontology.service.spec.ts#L82-L86

SO when you expect an HTTP request to have been sent, you check that the URL and method is correct and send back the mocked data.